### PR TITLE
Bruce microscope focus map

### PR DESCRIPTION
**Pull request by andrewbrucelau-zaber**

### CODE REVIEW

File: src/microscope_focus_map/least_square_interpolation.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+Assuming we have a number of points $(x, y, z)$ where $x$ and $y$ are the Cartesian coordinates of the position on the microscope slide, and $z$ is the focus height.  The simplest way to think about it is to fit a plane to the points.
+
+The equation of the plane is $ax + by + c = z$.
+
+Since there are three unknown coefficients ($a$, $b$, and $c$), we would need three equations to solve for the unknowns.  This is the same as saying if we have three points $(x_n, y_n, z_n)$, we can deterministically define the plane (assuming the three points form a triangle instead of a line).
+
+$$
+    ax_0 + by_0 + c = z_0 \\
+    ax_1 + by_1 + c = z_1 \\
```
[the remark this answers is not in the document]
Comment (andrewbrucelau-zaber) [reply src/microscope_focus_map/least_square_interpolation.md]: Thanks, fixed.

File: src/microscope_focus_map/least_square_interpolation.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
+# Least Square Interpolation
+This article explains the math behind the example code for focus map.  We will start with the simplest case of fitting a plane, then extend the idea to doing bilinear and higher order interpolation.
+
+## Fitting a Plane with 3 points
+
+Assuming we have a number of points $(x, y, z)$ where $x$ and $y$ are the Cartesian coordinates of the position on the microscope slide, and $z$ is the focus height.  The simplest way to think about it is to fit a plane to the points.
```
[the remark this answers is not in the document]
Comment (andrewbrucelau-zaber) [reply src/microscope_focus_map/least_square_interpolation.md:6]: Thanks, adopted your suggested language.

File: src/microscope_focus_map/least_square_interpolation.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+The equation of the plane is $ax + by + c = z$.
+
+Since there are three unknown coefficients ($a$, $b$, and $c$), we would need three equations to solve for the unknowns.  This is the same as saying if we have three points $(x_n, y_n, z_n)$, we can deterministically define the plane (assuming the three points form a triangle instead of a line).
```
[the remark this answers is not in the document]
Comment (andrewbrucelau-zaber) [reply src/microscope_focus_map/least_square_interpolation.md:10]: Done.  used p=0,1,2.

File: src/microscope_focus_map/least_square_interpolation.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+Reference: [Best Fitting Plane Given a Set of Points on Stackexchange](https://math.stackexchange.com/questions/99299/best-fitting-plane-given-a-set-of-points#answer-2306029)
+
+## Bilinear Interpolation
+
+The example above to fit a plane can be easily extended to the case of Bilinear Interpolation.  Bilinear Interpolation is like a plane, but with 4 points instead of 3.  As a result it is more akin to having a warped plane or stretched fabric pinned at four corners.
+
+Reference: [Bilinear Interpolation on Wikipedia](https://en.wikipedia.org/wiki/Bilinear_interpolation)
+
+If we were to rewrite the formula of the Plane and formula of Bilinear Interpolation side-by-side, they would look very similar except Bilinear Interpolation has an extra term.
```
[the remark this answers is not in the document]
Comment (andrewbrucelau-zaber) [reply src/microscope_focus_map/least_square_interpolation.md:141]: done.

File: src/microscope_focus_map/least_square_interpolation.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+        a_{00}\\
+        \vdots\\
+        a_{22}
+    \end{bmatrix}
+    = (A^TA)^{-1}A^TB
+$$
+
+### Bicubic and higher order interpolation
+
+Bicubic Interpolation is simply extending the above to n=3, and the interpolation formula has 4 x 4 = 16 coefficients.
```
[the remark this answers is not in the document]
Comment (andrewbrucelau-zaber) [reply src/microscope_focus_map/least_square_interpolation.md:293]: Fixed as discussed.

**Review by daryl-zaber — APPROVED** [review]

**Pull request merged**